Move asset pipeline javascripts to frontend

oliverguenther opened the pull request:
### ToDo

- [x] Move global JS files to frontend/src/...
  - [x] Get frontend assets green (npm run serve)
- [x] Remove/Replace .js.erb files
- [x] Remove obsolete .js files
- [x] Load all plugin .js files correctly
  - [x] Plugins which are already under frontend/modules
  - [x] Subscription plugin (done in https://github.com/finnlabs/saas-openproject/pull/74)
- [x] Finish all Todos (marked like that: Todo for task #33507)

oliverguenther committed:
ove reporting and reporting_engine js to frontend page

oliverguenther committed:
Replace webhook assets javascript

oliverguenther committed:
Add path augment service to dynamically load scripts

HDinger commented:
Ticket:
https://community.openproject.com/projects/openproject/work_packages/33507/activity

HDinger committed:
Remove JS-Files that are not used/not needed any more

HDinger committed:
Include AugmentingModule correctly

HDinger committed:
Remove special handling for checkbox selection which is outdated and two jquery libs that are not used any more

HDinger committed:
Load scripts dynamically that are not needed globally

openprojectci commented on frontend/src/app/modules/reporting/reporting-page/reporting-page.component.ts:
```diff
@@ -1,7 +1,15 @@
+import './functionality/reporting_engine/restore_query';
+import './functionality/reporting_engine/controls';
+
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/7c43ca799794047bcf272d9ecda248b84d763c76/frontend/src/app/modules/reporting/reporting-page/reporting-page.component.ts#L12"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci replied:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/743dd4d4e3903df104c4b7f27c4d62f6a5c6bdc8/frontend/src/app/modules/reporting/reporting-page/reporting-page.component.ts#L12"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

oliverguenther committed:
Move plugin JS to frontend

oliverguenther committed:
Move toggable fieldsets to global listeners

oliverguenther committed:
Move top-menu to global listeners and convert to TS

oliverguenther committed:
Move action_menu to frontend and port to TS

oliverguenther committed:
Move settings listener to frontend

oliverguenther committed:
Remove global js assets that were not used anyway

oliverguenther committed:
Move date range to timelog dynamic script where it's still used until removed

oliverguenther committed:
Move colors to global-listeners and port to TS

oliverguenther committed:
Replace tooltip with simple form instructions on the one place used

Tooltips were only used at the one place in our application, when changing a users password.
We can simply remove it there and use a modern solution the next time we need a tooltip

oliverguenther committed:
Move danger zone validation to frontend and port to TS

oliverguenther committed:
Move colorcontrast to backlogs vendor where it's used now

oliverguenther committed:
Remove contents of application.js.erb

oliverguenther committed:
Move project and user scripts to dynamically loaded

oliverguenther committed:
Move onboarding tour to dynamically loaded

HDinger committed:
Use correct syntax for jQuery so that $ is  known keyword within this function

oliverguenther committed:
Fix onboarding tour

oliverguenther committed:
Fix onboarding tour for scrum tour

oliverguenther committed:
Fix specs after moving JS

openprojectci commented:
<!--
  0 Errors
  1 Warning: <a href="https://github.com/op...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/743dd4d4e3903df104c4b7f27c4d62f6a5c6bdc8/frontend/src/app/modules/admin/types/type-form-configuration.component.ts#L33">frontend/src/app/modules/admin/types/type-form-configuration.component.ts#L33</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

oliverguenther committed:
Remove application.js.erb

oliverguenther committed:
Move locales and ckeditor to the frontend to dynamically load

oliverguenther committed:
Remove bundles caches

These are no longer being used since angular cli

oliverguenther committed:
Ensure locales are exported before angular
